### PR TITLE
CI: Add cargo-c tests

### LINKED ISSUE #1383: Add C API build to CI
$subj

### PR DESCRIPTION
**Pull request by Luni-4**
This PR closes #1383

### CODE REVIEW

**Comment by lu-zero** [comment]
Something went wrong with the cache server though.

File: .github/workflows/rav1e.yml [hunk 1]
```diff
@@ -283,15 +302,19 @@ jobs:
     - name: Set environment variables
       run: |
         echo "::add-path::$Env:VS_PATH\$Env:COMPILER_SUBPATH;C:\nasm"
-    - name: Install Rust
+    - name: Install ${{ matrix.toolchain }}
```
Comment (lu-zero) [inline .github/workflows/rav1e.yml]: Is this related?

File: .github/workflows/rav1e.yml [hunk 2]
```diff
@@ -283,15 +302,19 @@ jobs:
...
       uses: actions-rs/toolchain@v1
       with:
         profile: minimal
-        toolchain: stable
+        toolchain: ${{ matrix.toolchain }}
```
Comment (lu-zero) [inline .github/workflows/rav1e.yml]: Same question here

**Comment by Luni-4** [comment]
> I like it, later would be great having an action to package the binary package for the windows users I guess.

Does the binary package added in the `deploy action` not enough? Or are you talking about a `chocolatey` package?

File: .github/workflows/rav1e.yml (hunk 1 above)
Comment (Luni-4) [reply .github/workflows/rav1e.yml]: Hmm, I don't think so, but I can investigate on it

File: .github/workflows/rav1e.yml (hunk 2 above)
Comment (Luni-4) [reply .github/workflows/rav1e.yml]: Same reply as above

**Comment by lu-zero** [comment]
> Does the binary package added in the `deploy action` not enough? Or are you talking about a `chocolatey` package?

As in having on one side the `.exe`, on the other the `.dll`, `.h` and `.pc`.

**Comment by Luni-4** [comment]
> As in having on one side the `.exe`, on the other the `.dll`, `.h` and `.pc`.

For now they are in two different `.zip` files, but I could create a single `.zip` file for `Windows` containing all the binaries. Do you like this idea?
